modules/internal/DefaultAssociative.chpl

```diff
@@ -566,14 +566,15 @@ module DefaultAssociative {
 
     // ref version
     proc dsiAccess(idx : idxType, haveLock = false) ref {
-      const shouldLock = dom.parSafe && !haveLock;
+      const arrOwnsDom = dom._arrs.length == 1;
+      const shouldLock = dom.parSafe && !haveLock && arrOwnsDom;
```

Comment:
Just to make sure I'm interpreting this correctly -- should I read your "What if" as "Do you think it could/should do this?"  (at first I read it as proposing a hypothetical situation that could be problematic).